remote_vector_index_builder/core/common/models/index_builder/faiss/faiss_gpu_index_cagra_builder.py

```diff
@@ -0,0 +1,214 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import faiss
+from typing import Dict, Any
+from dataclasses import field
+
+from core.common.models import (
+    CagraGraphBuildAlgo,
+    FaissGPUIndexBuilder,
+    FaissIndexIDMap,
+    VectorsDataset,
+    SpaceType,
+)
+from core.index_builder.index_builder_utils import configure_metric
+from .ivf_pq_build_cagra_config import IVFPQBuildCagraConfig
+from .ivf_pq_search_cagra_config import IVFPQSearchCagraConfig
+
+
+class FaissGPUIndexCagraBuilder(FaissGPUIndexBuilder):
+    """
+    Configuration class for Faiss GPU Index Cagra
+    Also exposes a method to build the gpu index from the CAGRA configuration
+    """
+
+    # Degree of input graph for pruning
+    intermediate_graph_degree: int = 128
```

Comment:
there is another parameter called as refinement, whose default value is 2. Lets use that  parameter too here. Please check this reference: https://github1s.com/facebookresearch/faiss/blob/main/faiss/gpu/GpuIndexCagra.h#L178-L179